specs/AppNotifications/AppNotificationContentSpec/AppNotificationBuilder-spec.md

```diff
@@ -0,0 +1,1170 @@
+</toast>
+```
+
+The ToolTip is useful for buttons with icons and no content. Users can hover over the button with
+the cursor and a text will display describing what the button does.
+
+![Button ToolTip Example](ButtonToolTipExample.png)
+
+AppNotificationContent code:
+
+```c#
+AppNotificationContent()
+    .AddText(L"Content")
+    .AddTextBox(L"textBox", L"Reply")
+    .AddButton(AppNotificationButton()
+        .AddArgument(L"action", L"textReply")
+        .SetInputId(L"textBox")
+        .SetToolTip(L"Send")
+        .SetIcon(winrt::Windows::Foundation::Uri(LR"(Assets/Icons/message.png)")))
+    GetXml();
+```
+
+XML output:
+
+```c#
+<toast>
+  <visual>
+    <binding template="ToastGeneric">
+      <text>Content</text>
+    </binding>
+  </visual>
+
+    <actions>
+            <input id="textBox" type="text" placeHolderContent="Reply"/>
+            <action
+            content=""
+            imageUri="Assets/Icons/message.png"
+            hint-inputId="textBox"
+            hint-toolTip="Send"
+            arguments="action=textReply"/>
+    </actions>
+</toast>
+```
+
+The placement attribute defines action for a user to interact with when the AppNotification is
+right-clicked. This would typically be used by developers to ask the user for setting changes.
+
+![Button ContextMenu Example](ButtonContextMenuExample.png)
+
+AppNotificationContent code:
+
+```c#
+AppNotificationContent()
+    .AddText(L"Content")
+    .AddTextBox(L"textBox", L"Reply")
+    .AddButton(Button(L"Modify app settings")
+        .AddArgument(L"action", L"textReply")
+        .UseContextMenuPlacement())
+    .GetXml();
+```
+
+XML output:
+
+```c#
+<toast>
+  <visual>
+    <binding template="ToastGeneric">
+      <text>Content</text>
+    </binding>
+  </visual>
+
+    <actions>
+            <input id="textBox" type="text" placeHolderContent="Reply"/>
+            <action
+            placement="contextMenu"
+            content="Modify app settings"
+            imageUri="Assets/Icons/message.png"
+            arguments="action=textReply"/>
+    </actions>
+</toast>
+```
+
+## Limitations
+
+-   hint-toolTip and hint-buttonStyle are only supported for builds 19041 (20H1) and above. Using
+    these attributes on lower builds will not prevent the AppNotification from being constructed,
+    but developers will receive a debug warning that the attributes are not supported.
+
+# Image
+
+The Image component API sets up the xml for the \<image\> element. Images are visual elements that
+are used to enhance an AppNotification.
+
+The image source can be pulled using one of these protocol handlers:
+
+-   http:// or https:// (A web-based image) (unavailable for unpackageds on build 22000)
+-   ms-appx:/// (An image included in the app package) (only for packaged applications)
+-   ms-appdata:///local/ (An image saved to local storage) (only for packaged applications)
+-   file:/// (A local image)
+
+**WinAppSDK 1.2 \<image\> Schema**:
+
+```c#
+<image src = string
```

Comment:
Our docs only list the protocols, I can't find a list of supported file types. Can you help me find these? @andrewleader

Reply:
@pmpurifoy I unfortunately don't have a list of supported files, you'll have to talk to Shell to get a list... I think they basically pass through their images to XAML, so whatever the XAML image type supports is what notifications support (so SVG is supported in newer-ish versions of Windows when XAML added SVG support to their Image control, for example). I don't think GIFs are supported since I don't think the XAML Image class supports GIF?

PNG, JPG should have been supported since day one. I think SVG was added sometime around 19041 or possibly a little later. You should be able to check with the XAML team too (since I think that's basically what the Shell team will do if you ask them). I appreciate you helping identify this list! Devs will appreciate your efforts!!

Reply:
png, jpg, and svg are supported. Will update spec.